### PR TITLE
feat(settings): Update AccountRecoveryConfirmKey to new design with hint

### PR DESCRIPTION
**Pull request by vpomerleau**
UI updated, needs to be hooked up with actual storage hint from `FXA-7400`

## Because

* We are updating the design of the `AccountRecoveryConfirmKey` page
* We want to show the storage hint that users have had the option of saving

## This pull request

* Update `AccountRecoveryConfirmKey` page content, l10n, stories, test
* Update styling of recovery key data block on download page where key was overflowing

## Issue that this pull request solves

Closes: #FXA-7894

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

image

image

image

## Other information (Optional)

I had initially updated the alignment on the reset password page as a bonus, but opted to remove that update since the copy might also need to be updated to align with the latest designs. I'll file a separate ticket for those changes.

Added a character counter - only counts letters numbers and ignores spaces that will be stripped out on submit.

### CODE REVIEW

File: packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx [hunk 1]
```diff
@@ -74,46 +76,56 @@ const AccountRecoveryConfirmKey = ({
...
 
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (errorMessage) {
+      setErrorMessage('');
+    }
+    // strip out any characters other than letters, numbers or spaces
+    // spaces are left in for ease of user review but removed on submit
+    const filteredCode = e.target.value.replace(/[^a-zA-Z0-9\s]/g, '');
+    e.target.value = filteredCode;
+    e.target.value.replace(/\s/g, '').length === 32
```
Comment (chenba) [inline packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx:90]: Why the second `replace`?  The spaces were remove in the first `replace` call.  nit: `filteredCode` seems unnecessary; could just directly assign the return value of teh first `replace` into `e.target.value`.

**Review by chenba — APPROVED** [review]
👍 LGTM.  Just the one nit comment.

File: packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx (hunk 1 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx:90]: I'll see if I can make this clearer - the first `replace` filters out anything other than alphanumeric or spaces, but the second removes the spaces to check if the code has a valid format. Agree on the second point though.

Comment (chenba) [reply packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx:90]: Oh haha yeah forgot about the `^`.  But why not replace the spaces also?

Comment (chenba) [reply packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx:90]: Sorry I feel like I was unintentionally asking a simple question in a roundabout fashion... so if we want to check if there are 32 alphanumeric characters in the input, then why not directly go there?

Comment (vpomerleau) [reply packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx]: The intent was for the user to be able to see the spaces if they're copying the code (easier for them to review) but ignore them when checking whether or not to enable submission. Certainly easier to do away with that and just strip out the spaces. I'll update that way and ask Eduardo to take a look at the storybook.

Comment (chenba) [reply packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryConfirmKey/index.tsx]: > The intent was for the user to be able to see the spaces if they're copying the code (easier for them to review) but ignore them when checking whether or not to enable submission.

Ah, gotcha.  That make sense.  Thank you for the explanation.

**Review by bcolsson — APPROVED** [review]

**Pull request merged**